tests/test_target_storage_types.py

```diff
@@ -0,0 +1,592 @@
+        package2 = Package.get_package('test_package2', create=True)
+
+        create_info = [{'product': self.apple, 'qty': 4}]
+        picking = self.create_picking(self.picking_type_in,
+                                      products_info=create_info,
+                                      confirm=True)
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 2}]
+        self.validate_move_lines(picking.move_line_ids, self.apple, 4, 0)
+
+        self.picking_type_in.u_target_storage_format = 'package'
+        picking.update_picking(products_info=products_info, result_package_name=package1.name)
+        apple_lines_done = picking.move_line_ids.filtered(lambda x: x.product_id == self.apple and x.qty_done > 0)
+        self.validate_move_lines(apple_lines_done, self.apple, 2, 2, is_package=True, package_name=package1.name)
+
+        picking.update_picking(products_info=products_info, result_package_name=package2.name)
+        apple_lines_done2 = picking.move_line_ids.filtered(lambda x: x.product_id == self.apple and x.qty_done > 0)
+        apple_lines_done2 -= apple_lines_done
+        self.validate_move_lines(apple_lines_done2, self.apple, 2, 2, is_package=True, package_name=package2.name)
+
+    def test18_target_storage_format_multiple_packages_from_same_product_pallet_products(self):
+        """Tests that mutiple sets of the same product can be made from a single picking
+           for target_storage_format pallet_products.
+        """
+        # Two packages from same product
+        Package = self.env['stock.quant.package']
+        package1 = Package.get_package('test_package1', create=True)
+        package2 = Package.get_package('test_package2', create=True)
+
+        create_info = [{'product': self.apple, 'qty': 4}]
+        picking = self.create_picking(self.picking_type_in,
+                                      products_info=create_info,
+                                      confirm=True)
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 2}]
+        self.validate_move_lines(picking.move_line_ids, self.apple, 4, 0)
+
+        self.picking_type_in.u_target_storage_format = 'pallet_products'
+        picking.update_picking(products_info=products_info, result_package_name=package1.name)
+        apple_lines_done = picking.move_line_ids.filtered(lambda x: x.product_id == self.apple and x.qty_done > 0)
+        self.validate_move_lines(apple_lines_done, self.apple, 2, 2, is_pallet=True, package_name=package1.name)
+
+        picking.update_picking(products_info=products_info, result_package_name=package2.name)
+        apple_lines_done2 = picking.move_line_ids.filtered(lambda x: x.product_id == self.apple and x.qty_done > 0)
+        apple_lines_done2 -= apple_lines_done
+        self.validate_move_lines(apple_lines_done2, self.apple, 2, 2, is_pallet=True, package_name=package2.name)
+
+    def test19_target_storage_format_multiple_packages_from_same_product_pallet_packages(self):
```

Comment:
We need the same test calling update 2 times but with the same package. We should end up with 2 move lines with different result_package_id but with same u_result_parent_package_id. Because in this test we end up with 2 move lines where both result_package_id and u_result_parent_package_id are different.